wakebox/main.cpp

```diff
@@ -98,3 +137,22 @@ int main(int argc, char *argv[])
 
 	return res;
 }
+
+char* escaped_command_env(const JAST& jast) {
+	std::stringstream ss;
+	ss << "WAKEBOX_CMD=";
+	for (auto &child : jast.get("command").children) {
+		const std::string& s = child.second.value;
+		ss << '"';
+		for (std::string::size_type i = 0; i < s.size(); i++) {
+			if (s[i] == '"')
+				ss << '\\';
+			ss << s[i];
+		}
+		ss << '"' << " ";
+	}
+	// Stack allocated buffers shouldn't be used for exec's env arg.
+	char* var_on_heap = (char*)malloc(ss.str().length()+1);
+	strcpy(var_on_heap, ss.str().c_str());
+	return var_on_heap;
```

Comment:
`ss` is stack-allocated, and freed on return.
So, you had to copy the value because otherwise you had dangling pointer.

Return a std::string instead and you won't have problems.

Reply:
Also, by using malloc() to work-around the stack-allocation issue, you are leaking memory.